lib/pure/streams.nim

```diff
@@ -241,19 +244,29 @@ proc readDataStr*(s: Stream, buffer: var string, slice: Slice[int]): int =
     # fallback
     result = s.readData(addr buffer[0], buffer.len)
 
-when not defined(js):
+template jsOrVm(caseJsOrVm, caseElse: untyped): untyped =
+  when nimvm:
+    block:
+      caseJsOrVm
+  else:
+    when defined(js):
```

Comment:
```suggestion
    when defined(js) or defined(nimscript):
```
IIRC `when nimvm` is not defined for nimscript.

Reply:
not true:
```nim
proc fun() =
  when nimvm:
    echo "ok1"
  else:
    echo "ok2"
static: fun()
fun()
```
prints:
ok1
ok1

as expected

Reply:
This was a correct change, `moveMem`/`copyMem` are undeclared in NimScript but not in compile time VM, semantically checking `caseElse`